homeassistant/components/tado/binary_sensor.py

```diff
@@ -49,7 +51,9 @@ class TadoBinarySensorEntityDescription(BinarySensorEntityDescription):
 CONNECTION_STATE_ENTITY_DESCRIPTION = TadoBinarySensorEntityDescription(
     key="connection state",
     translation_key="connection_state",
-    state_fn=lambda data: data.get("connectionState", {}).get("value", False),
+    state_fn=lambda data: data.get("connectionState", {}).get(
```

Comment:
I'm thinking maybe rather make 2 entity descriptions, one for X one for V3, and only add them accordingly